src/engine/TransitivePathBinSearch.cpp

```diff
@@ -0,0 +1,245 @@
+// Copyright 2019, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Florian Kramer (florian.kramer@neptun.uni-freiburg.de)
+
+#include "TransitivePathBinSearch.h"
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <utility>
+
+#include "engine/CallFixedSize.h"
+#include "engine/TransitivePathBase.h"
+#include "util/Exception.h"
+#include "util/Timer.h"
+
+// _____________________________________________________________________________
+TransitivePathBinSearch::TransitivePathBinSearch(
+    QueryExecutionContext* qec, std::shared_ptr<QueryExecutionTree> child,
+    const TransitivePathSide& leftSide, const TransitivePathSide& rightSide,
```

Comment:
pass the `leftSide and right side and child by value and `std::move()` them. 
(Default for parameters that always need a copy).

Reply:
(Also check for other occurences of this pattern to make this a habit).